lib/raxx/context.ex

```diff
@@ -0,0 +1,110 @@
+defmodule Raxx.Context do
+  @type section_name :: term()
+
+  @typedoc """
+  An opaque type for the context snapshot data.
+  """
+  @opaque snapshot :: map()
+
+  @moduledoc """
+  `Raxx.Context` is a mechanism for simple sharing of state/information between
+  `Raxx.Middleware`s and `Raxx.Server`s.
+
+  It is designed to be flexible and to enable different middlewares to operate
+  on it without conflicts. Each separate functionality using the context
+  can be in a different "section", containing arbitrary data.
```

Comment:
I don't think that "sections" were implemented in the end?

Reply:
The individual section manipulation functions are gone, but I think a "section" is still a useful concept. It makes it more obvious that the context is not one blob of data, but a collection of sections that serve different purposes.

I could very well see it used in documentation as something like "The values stored in the session are available in the `MyLibrary.Session.Assigns` `Raxx.Context` section."